parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputCommitter.java

```diff
@@ -49,29 +50,48 @@ public void commitJob(JobContext jobContext) throws IOException {
   }
 
   public static void writeMetaDataFile(Configuration configuration, Path outputPath) {
-    if (configuration.getBoolean(ParquetOutputFormat.ENABLE_JOB_SUMMARY, true)) {
+    JobSummaryLevel level = ParquetOutputFormat.getJobSummaryLevel(configuration);
+
+    if (level != JobSummaryLevel.NONE) {
       try {
         final FileSystem fileSystem = outputPath.getFileSystem(configuration);
         FileStatus outputStatus = fileSystem.getFileStatus(outputPath);
-        List<Footer> footers = ParquetFileReader.readAllFootersInParallel(configuration, outputStatus);
+        List<Footer> footers;
+
+        if (level == JobSummaryLevel.ALL) {
+          footers = ParquetFileReader.readAllFootersInParallel(configuration, outputStatus, false); // don't skip row groups
+        } else if (level == JobSummaryLevel.COMMON_ONLY) {
+          footers = ParquetFileReader.readAllFootersInParallel(configuration, outputStatus, true); // skip row groups
+        } else {
+          throw new IllegalArgumentException("Unrecognized job summary level: " + level);
+        }
+
         // If there are no footers, _metadata file cannot be written since there is no way to determine schema!
         // Onus of writing any summary files lies with the caller in this case.
         if (footers.isEmpty()) {
           return;
         }
+
         try {
-          ParquetFileWriter.writeMetadataFile(configuration, outputPath, footers);
+          ParquetFileWriter.writeMetadataFile(configuration, outputPath, footers, level);
         } catch (Exception e) {
```

Comment:
Sounds like a good idea to me, but I assumed this was intentional (don't fail the entire job because the metadata couldn't be written?)

This catches Exception, it wouldn't catch an OOM which is an Error right? 

We can make failures here fatal, but I think we'd have to discuss with the community to see if anyone cares about that behavior.